### PR TITLE
Add light edges to extended model (SYN-6976)

### PR DESCRIPTION
**Pull request by Cisphyx**

### CODE REVIEW

File: synapse/datamodel.py [hunk 1]
```diff
@@ -1109,3 +1119,6 @@ def univ(self, name):
 
     def tagprop(self, name):
         return self.tagprops.get(name)
+
+    def edge(self, name):
```
Comment (invisig0th) [inline synapse/datamodel.py:1123]: Is `name` here really an `edgetype` tuple?

File: synapse/cortex.py [hunk 2]
```diff
@@ -3573,6 +3594,64 @@ async def _delTagProp(self, name):
...
+            return
+
+        self.model.delEdge(edge)
+
+        (n1form, verb, n2form) = edge
+
+        self.extedges.pop(edgeguid, None)
+        await self.fire('core:extmodel:change', n1form=n1form, verb=verb, n2form=n2form, act='del', type='edge')
+
+        edgedef = {'n1form': n1form, 'verb': verb, 'n2form': n2form}
```
Comment (invisig0th) [inline synapse/cortex.py:3652]: Could we just send along an `edgetype` tuple here (and above)? I'd like to start trying to make these messages a bit more compact :)

File: synapse/lib/stormlib/modelext.py [hunk 3]
```diff
@@ -77,6 +77,27 @@ class LibModelExt(s_stormtypes.Lib):
                       {'name': 'model', 'type': 'dict', 'desc': 'A model dictionary from getExtModel().', },
                   ),
                   'returns': {'type': 'boolean'}}},
+        {'name': 'addEdge', 'desc': 'Add an extended edge definition to the data model.',
```
Comment (invisig0th) [inline synapse/lib/stormlib/modelext.py:80]: Should probably specify here ( or in the args help ) that the verb must begin with _

**Review by invisig0th — CHANGES_REQUESTED** [review]

**Review by invisig0th — DISMISSED** [review]

**Review by Cisphyx — COMMENTED** [review]
Do we like `[\w\.:]+` (any unicode word characters, `.`, and `:`) and "must be lowercase" as the edge verb restrictions?

**Comment by invisig0th** [comment]
> Do we like `[\w\.:]+` (any unicode word characters, `.`, and `:`) and "must be lowercase" as the edge verb restrictions?

I think underscore is fine as well. What do we use in the grammar?

**Comment by Cisphyx** [comment]
> I think underscore is fine as well. What do we use in the grammar?

Underscore actually counts as a word character so that is included. Currently the grammar has no restrictions on the characters used in light edge verbs. This is basically a slightly looser regex than what we use for tags, `(\w+\.)*\w+`

**Review by invisig0th — APPROVED** [review]

**Pull request merged**